app/src/main/java/com/example/bottomnav/Statement.java

```diff
@@ -0,0 +1,116 @@
+package com.example.bottomnav;
+
+import org.eclipse.cdt.core.dom.ast.ExpansionOverlapsBoundaryException;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTDeclarator;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTEqualsInitializer;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTLiteralExpression;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTNamedTypeSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclaration;
+
+import java.util.Optional;
+
+// Only difference between const and members: type qualifiers and values. Parsing is the same.
+public abstract class Statement {
+    protected Optional<VariableContents> variableContents;
+    protected Optional<CPPASTDeclarator> declarator;
+    protected Optional<CPPASTEqualsInitializer> initializer;
+    protected Optional<CPPASTSimpleDeclSpecifier> simpleSpecifier;
+    protected Optional<CPPASTNamedTypeSpecifier> namedTypeSpecifier;
+
+    // Initializer function must check if components of the statement are null or not before
+    // extracting the string values, else parser may break.
+    public Statement(CPPASTSimpleDeclaration declaration) {
+        CPPASTDeclarator decl = (CPPASTDeclarator) declaration.getDeclarators()[0];
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) decl.getInitializer();
+
+        if (decl != null) {
+            declarator = Optional.of(decl);
+        } else {
+            declarator = Optional.empty();
+        }
+        if (init != null) {
+            initializer = Optional.of(init);
+        } else {
+            initializer = Optional.empty();
+        }
```

Comment:
```suggestion
        declarator = Optional.ofNullable(decl);
        initializer = Optional.ofNullable(init);
```
Check out Optional.ofNullable!